openapi/generator.go

```diff
@@ -273,7 +273,7 @@ func (g *Generator) AddOperation(path, method, tag string, in, out reflect.Type,
 	// Generate the default response from the tonic
 	// handler return type. If the handler has no output
 	// type, the response won't have a schema.
-	if err := g.setOperationResponse(op, out, strconv.Itoa(info.StatusCode), tonic.MediaType(), info.StatusDescription, info.Headers); err != nil {
+	if err := g.setOperationResponse(op, out, strconv.Itoa(info.StatusCode), tonic.MediaType(), info.StatusDescription, info.Headers, nil, nil); err != nil {
```

Comment:
Since we're adding a default response basec on tonic informations (status code), and the fact that it cannot be overriden, we should generate an example, if possible, or allow for default response override.

Reply:
For default responses, we currently do support generating examples; but only from the `example` tags on the struct itself.

For example, check this (see `examples/market/router.go`, changed in this PR):
```golang
	// Add a new fruit to the market.
	grp.POST("", []fizz.OperationOption{
		fizz.Summary("Add a fruit to the market"),
		fizz.Response("400", "Bad request", nil, nil,
			map[string]interface{}{"error": "fruit already exists"},
		),
	}, tonic.Handler(CreateFruit, 200))
```

which renders as:

![Screenshot 2020-11-24 at 12 56 57](https://user-images.githubusercontent.com/713026/100091202-995be880-2e54-11eb-85d6-24a77b61fb6f.png)

But I think I see what you had in mind. Currently, one cannot provide multiple examples for the default response nor is it able to provide an example with `fizz.Response`, but is forced to use the `example` tags on structs.

If we would want to support both of these, we would need to allow some sort of rewrites, which I'm a bit hesitant about. Unless we only allow the rewrites of examples but not other fields (i.e. something like `fizz.OverwriteExample(code int, example interface{})` and `fizz.OverwriteExamples(code int, examples map[string]interface{})`, but that then makes providing examples through the fizz.Response redundant)?

Do you think this is necessary, or could we leave it for future improvements?

Reply:
Yes, the examples values comes from the example of each fields individually. Overriding the default response could be introduced in the future with a separate function `WithDefaultResponse`. Lets leave it as it is.